e2e/consulcompat/run_ce_test.go

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package consulcompat
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	consulapi "github.com/hashicorp/consul/api"
+	"github.com/hashicorp/nomad/helper/uuid"
+	"github.com/hashicorp/nomad/testutil"
+	"github.com/shoenig/test/must"
+)
+
+func testConsulBuildLegacy(t *testing.T, b build, baseDir string) {
+	t.Run("consul-legacy("+b.Version+")", func(t *testing.T) {
+		cStop, consulHTTPAddr, consulAPI := startConsul(t, b, baseDir, "")
+		t.Cleanup(func() { cStop() })
+
+		// smoke test before we continue
+		verifyConsulVersion(t, consulAPI, b.Version)
+
+		// we need an ACL policy that allows the Nomad agent to fingerprint
+		// Consul, register services, render templates, and mint new SI tokens
+		consulToken := setupConsulACLsForServices(t, consulAPI,
+			"./input/consul-policy-for-nomad-legacy.hcl")
+
+		// we need service intentions so Connect apps can reach each other
+		setupConsulServiceIntentions(t, consulAPI)
+
+		// note: Nomad needs to be live before we can setupConsul because we
+		// need it up to serve the JWKS endpoint
+
+		consulCfg := &testutil.Consul{
+			Name:    "default",
+			Address: consulHTTPAddr,
+			Auth:    "",
+			Token:   consulToken,
+		}
+
+		nStop, nc := startNomad(t, consulCfg)
+		t.Cleanup(nStop)
+
+		verifyConsulFingerprint(t, nc, b.Version, "default")
+		runConnectJob(t, nc)
```

Comment:
Should we gate this on Linux and maybe have a non-Connect test for other systems?

Reply:
Yeah agreed we should gate it on Linux. I'm not sure it's worth doing a separate test for non-Linux. There's no OS-dependent behaviors other than the ones we just don't support at all (Connect).